Stop implying a build is required, and document the pull option (#923)
moderneinc/moderne-docs: docs/administrator-documentation/moderne-platform/how-to-guides/configuring-telemetry-exports/aws-replication.md

This guide walks you through receiving telemetry into an S3 bucket in **your** AWS account via S3 Cross-Region Replication (also works same-region). Before starting, read the [overview](./overview.md) for context on what the data looks like and how it flows.

:::tip

> If your policy forbids external principals writing into your account, or your bucket requires a customer-managed KMS key that Moderne's role cannot use, replication cannot be configured. See [Alternative: pull instead of push](#alternative-pull-instead-of-push) below.

Reviewer: "or your bucket requires a customer-managed KMS key that Moderne's role cannot use" --> Possibly being a bit pedantic, but this makes it sound like the problem is with Moderne not being able to use the KMS key when technically the barrier to being able to use it existed on the customer side because of company policy.

```

## Alternative: pull instead of push

Replication requires Moderne to write into your bucket. If your policy forbids that, you can pull instead: an IAM principal you own assumes a read-only role in Moderne's account and reads your tenant's telemetry on your own schedule. **No KMS key is shared in either direction**, and nothing external ever writes into your account.

The usual reason to need this is encryption policy. If your destination bucket requires a customer-managed SSE-KMS key and your security policy forbids external principals from using your keys, Moderne's replication role can never write an encrypted object, so replication cannot be configured at all.

### Setting up a pull

1. **Create the reading principal.** Make an IAM role in your account that will assume Moderne's pull role. Moderne's role ARN follows a fixed pattern, so you can reference it before it exists: `arn:aws:iam::<moderne-account-id>:role/moderne-bi-telemetry-pull-role-<your-tenant>`. Ask your CSM for the account ID if change control needs the full ARN up front.
2. **Send your CSM the principal ARN** along with your tenant name. Send more than one if different environments read separately.

> 3. **Moderne provisions the role** and sends back the role ARN to assume, the source bucket (`moderne-bi-telemetry-<your-tenant>`), and the key prefix (`tenant=<your-tenant>/`). The role grants `s3:ListBucket`, `s3:GetObject`, and `s3:GetBucketLocation` on that bucket only. It cannot write, delete, or reach another tenant's data.

Reviewer 2: 🤖 :
This says the source bucket is `moderne-bi-telemetry-<your-tenant>` (repeated in the command at line 120), but line 19 and line 84 of this same page, plus the diagram in `overview.md`, all say the bucket is `moderne-bi-telemetry` with tenant isolation done by the `tenant=<your-tenant>/` prefix:

> your tenant's IAM/RBAC only grants access to the `tenant=<your-tenant>/` prefix

At most one can be right, and a `tenant=` prefix inside an already-per-tenant bucket is the tell. My guess is the `-<your-tenant>` suffix got carried down from the role name on line 113, where it is correct. If so, both line 115 and line 120 should read `moderne-bi-telemetry`.

Run as written, the reader gets `NoSuchBucket`.

4. **Verify and schedule.** Assume the role, confirm you can list objects, then run a sync on a schedule:

```bash
aws s3 sync \
    s3://moderne-bi-telemetry-<your-tenant>/tenant=<your-tenant>/ \

>     s3://<your-bucket>/<your-prefix>/

Reviewer 2: 🤖 :
The destination here drops the `tenant=` key. `aws s3 sync` copies keys relative to the source prefix, so with source `.../tenant=acme/` the object `tenant=acme/source=saas/type=run/year=2026/.../abc123.csv` lands as `<your-prefix>/source=saas/type=run/year=2026/.../abc123.csv`, with no `tenant=` level.

That contradicts the **Keep the key layout** bullet at line 126, and it breaks the downstream examples this guide sends readers to:

* The DuckDB glob in `querying-and-bi.md` (`s3://<your-dest-bucket>/tenant=acme/*/type=run/**/*.csv`) matches nothing.
* The Databricks `PARTITIONED BY (tenant, source, type, year, month, day)` table has no `tenant` key to bind.
* This page's own verification command at line 95 (`aws s3 ls "s3://$DEST_BUCKET/tenant=$YOUR_TENANT/"`) returns empty.

Push and pull would silently produce different layouts.

Fix: make the destination `s3://<your-bucket>/<your-prefix>/tenant=<your-tenant>/`.

```

Two things to get right as you copy:

> 

Reviewer 2: ```suggestion
```

I don't think this line adds anything so would suggest removing it.

* **Keep the key layout.** The `tenant=`/`source=`/`type=`/`year=`/`month=`/`day=` structure is what lets query engines prune partitions. Flattening it means every query scans everything.
* **Run it on a schedule.** A pull only captures what is in the source bucket when it runs, so schedule it regularly rather than on demand.

### Push versus pull

> | | Replication (push) | Pull |

Reviewer 2: If you could run a format on these docs in VS code (or maybe even Claude can do it) that would be swell.

|---|---|---|
| Who moves the data | Moderne, continuously | You, on your schedule |
| Lands in your account | Automatically, within ~15 minutes | When your job next runs |
| Your KMS key | Moderne's role must be able to use it | Never shared |
| Inbound access to your account | Required | None |
| Missed data if your side breaks | Replication retries | Your job has to catch up on its next run |

> Replication is the better default wherever it's permitted. Use pull when it isn't.

Reviewer 2: ```suggestion
Use replication where your policy allows it, and pull where it doesn't.
```
